src/main/java/com/thebois/models/beings/roles/IdleRole.java

```diff
@@ -1,13 +1,41 @@
 package com.thebois.models.beings.roles;
 
+import java.util.Collection;
+import java.util.List;
+
+import com.thebois.models.beings.actions.ActionFactory;
+import com.thebois.models.beings.actions.IActionGenerator;
+import com.thebois.models.world.IWorld;
+
 /**
  * The builder constructs buildings.
  */
 class IdleRole extends AbstractRole {
 
+    private final IWorld world;
+
+    /**
+     * Instantiates with a world to randomly move around in.
+     *
+     * @param world The world to move around in.
+     */
+    IdleRole(final IWorld world) {
+        this.world = world;
+    }
+
     @Override
     public RoleType getType() {
         return RoleType.IDLE;
     }
 
+    @Override
+    protected Collection<IActionGenerator> getTaskGenerators() {
+        final IActionGenerator randomMove = getRandomMove();
+        return List.of(randomMove);
+    }
+
+    private IActionGenerator getRandomMove() {
+        return performer -> ActionFactory.createMoveTo(world.getRandomVacantSpot().getPosition());
```

Comment:
Does `World` have a `getRandomVacantSpot()`? It's not wrong but feels sort of weird as well, don't you agree? The other way would be either the Action or Pathfinder would generate a random pos and ask if its vacant. But might not be as elegant in reality as it sounds in my head atm. What is your opinion?

Reply:
It was the way you describe, before. 

The reason I changed it is because the responsibility of finding an empty position in the `World` is not `IdleRole`'s. It's `World`'s. Why? It's easy to notice when you write out how it would look if it weren't done in `World`:
```java
Position randomPos;
do{
    randomPos = generateRandomPosition();
} while(!world.isVacant(randomPos));
```
It's a violation of the law of demeter. `IdleRole` shouldn't repeatedly query the world, it should just ask the world for what the world knows. Since the world knows everything about what is inside of it, then only world should do the work of finding a random vacant spot.